### PR TITLE
Feature/externalize model list

### PR DESCRIPTION
**Pull request by vinodkiran**
Moving all model names to an external configuration (JSON) file.

### CODE REVIEW

**Comment by HenryHengZJ** [comment]
Related issue - https://github.com/FlowiseAI/Flowise/issues/1987

**Comment by HenryHengZJ** [comment]
approach looks good!

**Comment by HenryHengZJ** [comment]
new `gpt-4-turbo` released :)

File: packages/components/src/modelLoader.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { INodeOptionsValue } from './Interface'
+import axios from 'axios'
+
+/* This is a temp URL, will be replaced with the actual URL from the main repo */
+const MASTER_MODEL_LIST =
+    'https://raw.githubusercontent.com/vinodkiran/Flowise/FEATURE/externalize-model-list/packages/components/models.json'
```
Comment (HenryHengZJ) [inline packages/components/src/modelLoader.ts:6]: reminder to change this path

File: packages/components/src/modelLoader.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+export enum MODEL_TYPE {
+    CHAT = 'chat',
+    LLM = 'llm',
+    EMBEDDING = 'embedding'
+}
+
+const getModelConfig = async (category: MODEL_TYPE, name: string) => {
+    const modelFile = process.env.MODEL_LIST_CONFIG_JSON || MASTER_MODEL_LIST
```
Comment (HenryHengZJ) [inline packages/components/src/modelLoader.ts:15]: we need to add `MODEL_LIST_CONFIG_JSON` to other places like `start.ts` and oother `.env.example` files

File: packages/components/src/modelLoader.ts (hunk 1 above)
Comment (vinodkiran) [reply packages/components/src/modelLoader.ts:6]: Review and merge https://github.com/FlowiseAI/Flowise/pull/2153. Post that we can update this path to the updated config file.

Comment (HenryHengZJ) [reply packages/components/src/modelLoader.ts:6]: PR merged. we can update the path now

File: packages/components/src/modelLoader.ts (hunk 2 above)
Comment (vinodkiran) [reply packages/components/src/modelLoader.ts:14]: Done!

**Review by HenryHengZJ — APPROVED** [review]

**Pull request merged**